xknx/devices/device.py

```diff
@@ -80,3 +82,19 @@ async def do(self, action):
         """Execute 'do' commands."""
         # pylint: disable=invalid-name
         self.xknx.logger.info("Do not implemented action '%s' for %s", action, self.__class__.__name__)
+
+    def __eq__(self, other):
+        """Compare for quality."""
+        for key, value in self.__dict__.items():
+            if key == "device_updated_cbs":
+                continue
+            if key not in other.__dict__:
+                return False
+            if other.__dict__[key] != value:
+                return False
+        for key, value in other.__dict__.items():
+            if key == "device_updated_cbs":
+                continue
+            if key not in self.__dict__:
+                return False
+        return True
```

Comment:
```suggestion
    def __eq__(self, other):
        """Compare for equality."""
        return self.__dict__ == other.__dict__
```
should be alright. We will have to find out why the one callback is added twice and thus breaks the config-tests.

Reply:
The object generated from the config only had the device_update cb from devices.py (in the add method).

The other object we instantiated had the device_update_cb from the constructor and the device_update_db from the add method in devicey.py. Removing the constructor argument worked.